src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Extensions/DefaultTagHelperTargetExtension.cs

```diff
@@ -650,16 +651,47 @@ internal static string GetDeterministicId(CodeRenderingContext context)
         return uniqueId;
     }
 
-    private static string GetPropertyAccessor(DefaultTagHelperPropertyIntermediateNode node)
+    private static int GetPropertyAccessorLength(DefaultTagHelperPropertyIntermediateNode node)
+    {
+        var propertyAccessorLength =
+            node.FieldName.Length
+            + ".".Length
+            + node.PropertyName.Length;
+
+        if (node.IsIndexerNameMatch)
+        {
+            propertyAccessorLength +=
+                "[\"".Length
+                + (node.AttributeName.Length - node.BoundAttribute.IndexerNamePrefix.Length)
+                + "\"]".Length;
+        }
+
+        return propertyAccessorLength;
+    }
+
+    private static CodeWriter WritePropertyAccessor(CodeWriter writer, DefaultTagHelperPropertyIntermediateNode node)
     {
-        var propertyAccessor = node.FieldName + "." + node.PropertyName;
+        writer
+            .Write(node.FieldName)
+            .Write(".")
+            .Write(node.PropertyName);
```

Comment:
There's an overload of `CodeWriter.Write(...)` that takes an interpolated string handler.

```suggestion
        writer.Write($"{node.FieldName}.{node.PropertyName}");
```